components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -76,17 +76,18 @@ func (g *gpsrtk) closeCorrectionWriter() {
 		if err != nil {
 			g.logger.Errorf("Error closing port: %v", err)
 		}
+		g.correctionWriter = nil
```

Comment:
This is the line that prevents the panic when closing an already-closed correctionWriter. It matches line 296 down in `Close()`.